Added Treatment exercise draft (#4)
mrc-ide/malsimtutorial: inst/tutorials/treatment/treatment.Rmd

```

## 2. Loading antimalarial drug parameters

The second step is to load the parameters for the drug(s) we want to simulate into our list of parameters. The `malariasimulation` package has in-built parameter sets for 3 drugs:

1.  Artemether lumefantrine (`AL_params`)
2.  Dihydroartemisinin piperaquine (`DHA_PQP_params`)
3.  Sulfadoxine pyrimethamine (`SP_AQ_params`)

We can view these parameter sets by simply calling them. Here are the parameters for artemether lumefantrine:

```{r view-inbuilt-antimalarial-parameters, exercise = TRUE, exercise.eval = FALSE}

# View the artemether lumefantrine (AL) parameter set:
AL_params

```

The parameters in these sets are:

1.  drug_efficacy: probability that treatment with the drug successfully clears the infection
2.  drug_rel_c: The onward infectiousness of a treated individual to a mosquito relative to an untreated individual
3.  drug_prophylaxis_shape: Shape parameter for a Weibull distribution describing an individuals prophylactic protection over time following treatment
4.  drug_prophylaxis_scale: Scale parameter for a Weibull distribution describing an individuals prophylactic protection over time following treatment

> ```{r view-drug-params, exercise = TRUE}

Reviewer: Prompt them a bit more celawrly here what this code block is for

Author: I have added: "In the code chunk below, view the parameter sets for DHA-PQP and SP-AQ:"

# View the parameters for DHA-PQP and SP-AQ:

```

> ```{r drug-params-quiz, exercise = TRUE}

Reviewer: This isn't showing up as a quiz for me, perhaps:
```suggestion
```{r drug-params-quiz}
```

Author: Good catch - fixed!

quiz(
  question("The efficacy of DHA-PQP in the DHA_PQP_params set is 0.95000",
    answer("TRUE", correct = TRUE),
    answer("FALSE")
  )
)

```

To simulate the use of a drug in `malariasimulation`, we load them into the parameters list. This is done using the `set_drugs()` function. Note that the drug parameters (e.g. `AL_params`) must be passed to `get_drugs()` as a list.

```{r using-set-drugs, exercise = TRUE, eval = TRUE}

# Establish a list of malariasimulation parameters:
parameters <- get_parameters() 

# Load the parameters for artemether lumefantrine into the parameters
parameters <- set_drugs(parameters = parameters, drugs = list(AL_params))

> # View the parameters for DHA-PQP and SP-AQ:

Reviewer: ```suggestion
# View the parameters for AL:
```

Author: Good spot - corrected.

parameters$drug_efficacy

```

The final line of the chunk above shows us how we can view the drug parameters currently loaded the parameters list once they have been appended. When simulating interventions that use antimalarial drugs (e.g. clinical treatment), this can provide a useful check to ensure you have loaded the drug parameters correctly. In the chunk below, view the `drug_rel_c` parameter:

```{r view-drug-in-params, exercise = TRUE}

# Establish a list of malariasimulation parameters:
parameters <- get_parameters() 

# Load the parameters for artemether lumefantrine into the parameters
parameters <- set_drugs(parameters = parameters, drugs = list(AL_params))

> # View the drug_rel_c parameter for DHA-PQP:

Reviewer: Apols if I'm not following, but should these be AL?
```suggestion
# View the drug_rel_c parameter for AL:
```

Author: I think I'd meant to change the drug parameters loaded from AL to DHA-PQP - have fixed for now but thinking it ight be a bit repetitive so might take it out.

```

## 3. Setting clinical treatment

> Now that we know how to load antimalarial drug parameters we can specify how they will be administered to the population. In malariasimulation, we parameterise clinical treatment campaigns using the `set_clinical_treatment()` function (run `?set_clinical_treatment()` to view the help file for the function). `set_clinical_treatment()` requires four arguments:

Reviewer: I'd not call this a campaign, as that might mean something quite specific, and not usually associated with tx

Author: OK good to know! Have removed campaigns

```{r set-clin-treat-example, exercise = TRUE}

# Establish a list of malariasimulation parameters:
parameters <- get_parameters(overrides = list(human_population = 1000)) 

# Load the parameters for artemether lumefantrine into the parameters
parameters <- set_drugs(parameters = parameters, drugs = list(AL_params))

# Specify a clinical treatment regime for artemether lumefantrine:
parameters <- set_clinical_treatment(parameters = parameters, drug = 1, timesteps = c(0, (1 * 365) + 1), coverages = c(0, 0.6))

# Tune the parameter set to an initial EIR:
parameters <- set_equilibrium(parameters = parameters, init_EIR = 8)

# Run the simulation
clinical_treatment_simulation <- run_simulation(timesteps = 3 * 365, parameters = parameters)

# Plot the prevalence through time:
clinical_treatment_simulation |>
  ggplot(aes(x = timestep, y = n_detect_lm_730_3650/n_730_3650)) +
  geom_line(colour = "deepskyblue2", linewidth = 1.2) +
  theme_bw() +
  labs(x = "Time (days)", y = "Prevalence (2-10)")+
  scale_y_continuous(limits = c(0, 1), expand = c(0, 0)) +
  scale_x_continuous(expand = c(0, 0)) +
  geom_vline(xintercept = (1 * 365) + 1, linetype = "dashed")
  

```

> ```{r quiz2, exercise = TRUE}

Reviewer: As before- not showing up as a quiz for me

Author: Fixed

    answer(c(0.4, 0.5, 0.6)),
    answer(c(0, 0, 0)),
    answer(c(0.9, 0.4, 0.1)),
    answer(c(0, 0.5), correct = TRUE)
  )
)

```

## 4: Putting it all together

Awesome, that's all of the basics for running `malariasimulation` simulations with clinical treatment. For the final exercise, I want you to run two simulations to look at the effect of clinical treatment on malaria transmission dynamics.

The first simulation is a baseline scenario with the following criteria:

1.  `human_population` size of 10000
2.  Equilibrate the parameters to an initial EIR of 8 (refer the "basics" tutorial if you need to!)
3.  Run the simulation for 5 years

The second simulation will use what we've learnt to simulate clinical treatment with the following criteria:

1.  Set the `human_population` parameter to 10000
2.  Load the parameters for dihydroartemisinin piperaquine (DHA-PQP)
3.  Initiate clinical treatment with a coverage of 0.4 at the end of year 1
4.  Increase the coverage to 0.8 at the end of the third year
5.  Equilibrate the simulations with an initial EIR of 8
6.  Run both simulations for 5 years

> 7.  Plot the *Pf*PR[2-10] for each simulation and compare the outputs

Reviewer: I think this should format 2-10 as subscript
```suggestion
7.  Plot the *Pf*PR~2-10~ for each simulation and compare the outputs
```

Author: Done!

```{r load-malsim-parameters, exercise = TRUE, exercise.eval = FALSE}

# Establish a list of malariasimulation parameters:
parameters <- get_parameters(overrides = list(human_population = 1000))

```

## 2. Loading antimalarial drug parameters

The second step is to load the parameters for the drug(s) we want to simulate into our list of parameters. The `malariasimulation` package has in-built parameter sets for 3 drugs:

1.  Artemether lumefantrine (`AL_params`)
2.  Dihydroartemisinin piperaquine (`DHA_PQP_params`)
3.  Sulfadoxine pyrimethamine (`SP_AQ_params`)

We can view these parameter sets by simply calling them. Here are the parameters for artemether lumefantrine:

```{r view-inbuilt-antimalarial-parameters, exercise = TRUE, exercise.eval = FALSE}

# View the artemether lumefantrine (AL) parameter set:
AL_params

```

The parameters in these sets are:

1.  drug_efficacy: probability that treatment with the drug successfully clears the infection
2.  drug_rel_c: The onward infectiousness of a treated individual to a mosquito relative to an untreated individual

> 3.  drug_prophylaxis_shape: Shape parameter for a Weibull distribution describing an individuals prophylactic protection over time following treatment

Reviewer 2: Is it worth including my drug prophlyaxis plot through time here as a visual?
![image](https://github.com/mrc-ide/malsimtutorial/assets/73181123/fade0f7a-64f7-4ace-9f3b-ee2c59b5f960)
Then you could have a question about duration of prophylaxis maybe

Author: Added with a question about duration of prophylaxis!

4.  drug_prophylaxis_scale: Scale parameter for a Weibull distribution describing an individuals prophylactic protection over time following treatment

In the code chunk below, view the parameter sets for DHA-PQP and SP-AQ:

```{r view-drug-params, exercise = TRUE}

# View the parameters for DHA-PQP and SP-AQ:

```

```{r drug-params-quiz}
SP_AQ_params
quiz(
  question("The efficacy of DHA-PQP in the DHA_PQP_params set is 0.95000",
    answer("TRUE", correct = TRUE),
    answer("FALSE")
  ),
  question("The drug_rel_c of SP-AQ in the SP_AQ_params set is 4.30",
    answer("TRUE"),
    answer("FALSE", correct = TRUE)
  )
)

```

To simulate the use of a drug in `malariasimulation`, we load them into the parameters list. This is done using the `set_drugs()` function. Note that the drug parameters (e.g. `AL_params`) must be passed to `get_drugs()` as a list.

```{r using-set-drugs, exercise = TRUE, eval = TRUE}

# Establish a list of malariasimulation parameters:
parameters <- get_parameters() 

> # Load the parameters for artemether lumefantrine into the parameters

Reviewer 2: ```suggestion
# Load the drug parameters for artemether lumefantrine into the parameters list
```

Author: Good spot - changed!

Author: Fixed!

1.  `parameters`: A list of `malariasimulation` parameters (with parameters for drug(s) loaded using `set_drugs()`)
2.  `drug`: The indices of the drug which we are parameterising clinical treatment for
3.  `timesteps`: A vector of timesteps on which we want to update clinical treatment coverage
4.  `coverages`: A vector of clinical treatment coverages (proportion of the clinical infections that will be treated in each timestep).

Note that we need to call `set_clinical_treatment()` for each drug that we are specifying clinical treatment for, and that the `timesteps` and `coverage` inputs need to be vectors of equal length, such that each update to the coverage has a corresponding timestep on which the update should take place. The following chunk shows how to set up and run a simulation with clinical treatment:

```{r set-clin-treat-example, exercise = TRUE}

# Establish a list of malariasimulation parameters:
parameters <- get_parameters(overrides = list(human_population = 1000)) 

> # Load the parameters for artemether lumefantrine into the parameters

Reviewer 2: ```suggestion
# Load the parameters for artemether lumefantrine into the parameters:
```

Author: Added

parameters <- set_drugs(parameters = parameters, drugs = list(AL_params))

# Specify a clinical treatment regime for artemether lumefantrine:
parameters <- set_clinical_treatment(parameters = parameters, drug = 1, timesteps = c(0, (1 * 365) + 1), coverages = c(0, 0.6))

# Tune the parameter set to an initial EIR:
parameters <- set_equilibrium(parameters = parameters, init_EIR = 8)

> # Run the simulation

Reviewer 2: ```suggestion
# Run the simulation:
```

Author: Added

clinical_treatment_simulation <- run_simulation(timesteps = 3 * 365, parameters = parameters)

# Plot the prevalence through time:
clinical_treatment_simulation |>

>   ggplot(aes(x = timestep, y = n_detect_lm_730_3650/n_730_3650)) +

Reviewer 2: ```suggestion
  ggplot(aes(x = timestep, y = n_detect_730_3650/n_730_3650)) +
```

This update isn't in the default malsim yet!

Author: Weird - I don't know how it'd be working as I'm not in a malsim project? Have corrected though!

  geom_line(colour = "deepskyblue2", linewidth = 1.2) +
  theme_bw() +
  labs(x = "Time (days)", y = "Prevalence (2-10)")+
  scale_y_continuous(limits = c(0, 1), expand = c(0, 0)) +
  scale_x_continuous(expand = c(0, 0)) +
  geom_vline(xintercept = (1 * 365) + 1, linetype = "dashed")
  

```

```{r quiz2}

quiz(
  question("Drug parameters must be loaded into set_drugs() as a list?",
    answer("TRUE", correct = TRUE),

>     answer("FALSE")

Reviewer 2: ```suggestion
    answer("FALSE"),
    allow_retry = TRUE
```

Author: Added

  ),
  question("If we want to specify clinical treatment regimes for three drugs, how many times must we use set_clinical_treatment()?",
    answer("1", correct = TRUE),
    answer("2"),

>     answer("3"),

Reviewer 2: Shouldn't the answer be 3 here?

Author: it should and now is!

>     answer("4")

Reviewer 2: ```suggestion
    answer("4"),
    allow_retry = TRUE
```
Could be useful!

Author: Added!

  ),
   question("If we input the vector c(0, 365, 720) as the argument for timesteps into set_clinical_treatment(), which of the following vectors would NOT work for the coverages argument?",
    answer("c(0.4, 0.5, 0.6)"),
    answer("c(0, 0, 0)"),
    answer("c(0.9, 0.4, 0.1)"),

>     answer("c(0, 0.5)", correct = TRUE)

Reviewer 2: ```suggestion
    answer("c(0, 0.5)", correct = TRUE),
    allow_retry = TRUE
```

Author: Added

  )
)

```

## 4: Putting it all together

Awesome, that's all of the basics for running `malariasimulation` simulations with clinical treatment. For the final exercise, I want you to run two simulations to look at the effect of clinical treatment on malaria transmission dynamics.

The first simulation is a baseline scenario with the following criteria:

> 1.  `human_population` size of 10000

Reviewer 2: ```suggestion
1.  `human_population` size of 10,000
```

Author: I avoided adding the comma in case people copied/pasted and broke it

> 2.  Equilibrate the parameters to an initial EIR of 8 (refer the "basics" tutorial if you need to!)

Reviewer 2: ```suggestion
2.  Equilibrate the parameters to an initial EIR of 8 (refer the "introduction" tutorial if you need to!)
```

Sorry! I changed this back to introduction

Author: Changed

3.  Run the simulation for 5 years

The second simulation will use what we've learnt to simulate clinical treatment with the following criteria:

> 1.  Set the `human_population` parameter to 10000

Reviewer 2: ```suggestion
1.  Set the `human_population` parameter to 10,000
```

Author: Avoided doing this in case they try adding the comma in when setting the parameter

2.  Load the parameters for dihydroartemisinin piperaquine (DHA-PQP)

> 3.  Initiate clinical treatment with a coverage of 0.4 at the end of year 1

Reviewer 2: Is it worth putting these as %s?

Author: I thought about it but figured I'd keep it as similar to the input as possible for ease?

4.  Increase the coverage to 0.8 at the end of the third year
5.  Equilibrate the simulations with an initial EIR of 8

> 6.  Run both simulations for 5 years

Reviewer 2: Should 6 and 7 be a third section...? (Seems there's some redundancy)

Author: Have moved .7 to it's own line - wanted to keep the 5 years in each one for clarity

```{r using-set-drugs, exercise = TRUE, eval = TRUE}

# Establish a list of malariasimulation parameters:
parameters <- get_parameters() 

# Load the parameters for artemether lumefantrine into the parameters
parameters <- set_drugs(parameters = parameters, drugs = list(AL_params))

# View the parameters for AL:
parameters$drug_efficacy

```

The final line of the chunk above shows us how we can view the drug parameters currently loaded the parameters list once they have been appended. When simulating interventions that use antimalarial drugs (e.g. clinical treatment), this can provide a useful check to ensure you have loaded the drug parameters correctly. In the chunk below, view the `drug_rel_c` parameter:

```{r view-drug-in-params, exercise = TRUE}

# Establish a list of malariasimulation parameters:
parameters <- get_parameters() 

# Load the parameters for artemether lumefantrine into the parameters list:
parameters <- set_drugs(parameters = parameters, drugs = list(DHA_PQP_params))

> # View the drug_rel_c parameter for DHA-PQP:

Reviewer: Just a note, this line seems to be floating about here still
